client/components/bookmark.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+export default class Bookmark extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      saved: null
+    };
+    this.handleAdd = this.handleAdd.bind(this);
+  }
+
+  handleAdd() {
+    this.props.openDrawer();
+  }
+
+  render() {
+    if (this.props.route) {
```

Comment:
is `this.props.route` ever falsy? try to eliminate conditional logic if you can.